### PR TITLE
Add version 2018-02-01 Microsoft.ServiceFabric specification

### PR DESCRIPTION
**Pull request by juhacket**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by marstr** [comment]
Howdy @juhacket,

To get started, could you confirm that this new API Version (2018-02) will be based on the 2017-07 API Version? That will help me and @ravbhatnagar scope our review to just things that have changed since the previous version.

Also, I see that you're adding a number of suppressions. If you've already gotten approval from the ARM team, then we don't need to go any further. If you do not yet have approval for them, we can get that ball rolling too.

**Comment by juhacket** [comment]
@marstr - Yes, the new API version is based on 2017-07-01-preview. We have split up the specification as only the cluster resource has changed in this new version.

We have discussed the suppressions with @ravbhatnagar and he has approved them. The main issue is how the validation tooling handles proxy resources.

**Comment by juhacket** [comment]
@marstr - All changes have been made and are ready for review. Please notify before merge as we have to coordinate on our end.

**Comment by marstr** [comment]
@AutorestCI rebuild go

**Comment by marstr** [comment]
Sorry for the delay on the review, it's taking me a little extra time because of the refactor. I should have a review in by EOD tomorrow.

**Comment by juhacket** [comment]
No problem. The plan is to finalize the review and be ready to merge on Friday.

File: specification/servicefabric/resource-manager/readme.md
Comment (marstr) [inline specification/servicefabric/resource-manager/readme.md]: In general this PR looks pretty much textbook, and that is much appreciated!

I did want to point out for the sake of awareness, that referencing these common definitions will result in the clients calling the service with both API Versions, as seen here in the Go SDK:
https://github.com/Azure/azure-sdk-for-go/pull/1609/files#diff-97eaea7067d7811544da444ef9a815e8R351

If you're okay with that, feel free to acknowledge and I will approve this PR.

**Comment by ravbhatnagar** [comment]
This has been reviewed in person. Signing off!

File: specification/servicefabric/resource-manager/readme.md
Comment (marstr) [reply specification/servicefabric/resource-manager/readme.md]: Yep, that's what I mean. Thanks for confirming!

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
I'll plan on merging this Friday, is that correct @juhacket?

**Comment by juhacket** [comment]
@marstr Please delay the merge. We have some dependencies that are not in place yet.

**Comment by marstr** [comment]
Given the title change, I assume your dependencies are still not in place. I will wait for you, @juhacket, to ping this thread or send me an email before I merge.

**Pull request merged**
